src/Microsoft.ApplicationInsights/DataContracts/AvailabilityTelemetry.cs

```diff
@@ -178,6 +178,26 @@ public ITelemetry DeepClone()
             return new AvailabilityTelemetry(this);
         }
 
+        /// <summary>
+        /// Deeply clones the Extension of <see cref="AvailabilityTelemetry"/> object.
+        /// </summary>
+        IExtension IExtension.DeepClone()
```

Comment:
this is not ideal. `DeepClone` needs to return `ITelemetry` so you can pass it to the processors and initializers. So maybe `ITelemetry` should be implementing `IExtension`?